### PR TITLE
Add string type option in convert_row_to_numpy

### PR DESCRIPTION
**Pull request by hkvision**
Logically, only tf2 estimator and pytorch ray estimator can support string input. BigDL backends use Sample and string is not supported.

### CODE REVIEW

File: pyzoo/zoo/orca/learn/openvino/estimator.py [hunk 1]
```diff
@@ -111,7 +111,8 @@ def predict_transform(dict_data, batch_size):
                                               validation_data=None,
                                               feature_cols=feature_cols,
                                               label_cols=None,
-                                              mode="predict")
+                                              mode="predict",
+                                              accept_str_col=False)
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/openvino/estimator.py:115]: why do we need this?

Comment (hkvision) [reply pyzoo/zoo/orca/learn/openvino/estimator.py:115]: OpenVINO doesn't support string, no?

Comment (jason-dai) [reply pyzoo/zoo/orca/learn/openvino/estimator.py:115]: it defaults to `false`, no?

File: pyzoo/zoo/orca/learn/utils.py [hunk 2]
```diff
@@ -279,7 +280,7 @@ def _dataframe_to_xshards(data, feature_cols, label_cols=None):
 
 
 def dataframe_to_xshards(data, validation_data, feature_cols, label_cols, mode="fit",
-                         num_workers=None):
+                         num_workers=None, accept_str_col=True):
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/utils.py:283]: we should make it default to `false`, and only make it `true` for `TF2RayEstimator`

File: pyzoo/zoo/orca/learn/utils.py [hunk 3]
```diff
@@ -263,14 +263,15 @@ def process_xshards_of_pandas_dataframe(data, feature_cols, label_cols=None, val
         return data
 
 
-def _dataframe_to_xshards(data, feature_cols, label_cols=None):
+def _dataframe_to_xshards(data, feature_cols, label_cols=None, accept_str_col=True):
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/utils.py:266]: default to `false`

File: pyzoo/zoo/orca/learn/utils.py (hunk 2 above)
Comment (hkvision) [reply pyzoo/zoo/orca/learn/utils.py:283]: oh really, but as we discussed, dataframe to xshards logically can support any datatype so I think it is more reasonable to make default True?

**Comment by hkvision** [comment]
Added a dummy unit test @jason-dai

**Comment by hkvision** [comment]
http://10.239.176.111:18888/job/ZOO-PR-Python-Spark-2.4-py37-Horovod/2021/console

File: pyzoo/zoo/orca/learn/utils.py (hunk 2 above)
Comment (jason-dai) [reply pyzoo/zoo/orca/learn/utils.py:283]: I think we should only support string type for `TF2RayEstimator` at this moment; it makes sense to set it to false for all the other cases.

**Comment by jason-dai** [comment]
> Added a dummy unit test @jason-dai

It only tests the positive case, not the negative case (e.g., passing string for tf1 estimator); I think you should keep the original behavior (i.e., string not supported) for all cases except the tf2 estimaror

File: pyzoo/zoo/orca/learn/utils.py (hunk 2 above)
Comment (hkvision) [reply pyzoo/zoo/orca/learn/utils.py:283]: okay.

**Comment by jason-dai** [comment]
LGTM

**Comment by hkvision** [comment]
jenkins passed: http://10.239.176.111:18888/job/ZOO-PR-Validation/6117/

**Pull request merged**